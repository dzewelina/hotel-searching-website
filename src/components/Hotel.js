import React from 'react';
import PropTypes from 'prop-types';

const Hotel = ({ hotel }) => (
  <li className='hotel media'>
    <img className='align-self-start mr-3' src={hotel.Image} alt={`${hotel.Name}`} />
    <div className='media-body'>
      <h1 className='mt-0'>{hotel.Name}</h1>
      <span>{hotel.StarRating}<i className="fas fa-star"></i></span>
      <p>
        {hotel.Facilities.map((facility, i) => (<span key={i}>{facility} </span>))}
      </p>
    </div>
  </li>
);

Hotel.propTypes = {
  hotel: PropTypes.object.isRequired
};

export default Hotel;