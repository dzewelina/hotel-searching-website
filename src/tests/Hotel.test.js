import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
configure({ adapter: new Adapter() });
import renderer from 'react-test-renderer';

import React from 'react';
import Hotel from '../components/Hotel';

describe('Hotel component', () => {
  const testingHotel = {
    'Name': 'Testing Hotel',
    'StarRating': 3,
    'Facilities': ['car park', 'restaurant', 'bar'],
    'Image': 'https://images'
  };
  it('matches the snapshot', () => {
    const tree = renderer.create(<Hotel hotel={testingHotel}/>).toJSON();
    expect(tree).toMatchSnapshot();
  });
});