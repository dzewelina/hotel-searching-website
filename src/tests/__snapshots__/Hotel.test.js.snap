// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hotel component matches the snapshot 1`] = `
<li
  className="hotel media"
>
  <img
    alt="Testing Hotel"
    className="align-self-start mr-3"
    src="https://images"
  />
  <div
    className="media-body"
  >
    <h1
      className="mt-0"
    >
      Testing Hotel
    </h1>
    <span>
      3
      <i
        className="fas fa-star"
      />
    </span>
    <p>
      <span>
        car park
         
      </span>
      <span>
        restaurant
         
      </span>
      <span>
        bar
         
      </span>
    </p>
  </div>
</li>
`;

exports[`Hotels component matches the snapshot 1`] = `
<li
  className="hotel media"
>
  <img
    alt="Testing Hotel"
    className="align-self-start mr-3"
    src="https://images"
  />
  <div
    className="media-body"
  >
    <h1
      className="mt-0"
    >
      Testing Hotel
    </h1>
    <span>
      3
      <i
        className="fas fa-star"
      />
    </span>
    <p>
      <span>
        car park
         
      </span>
      <span>
        restaurant
         
      </span>
      <span>
        bar
         
      </span>
    </p>
  </div>
</li>
`;
